opentelekomcloud/services/ims/data_source_opentelekomcloud_images_image_v2.go

```diff
@@ -172,21 +169,31 @@ func dataSourceImagesImageV2Read(d *schema.ResourceData, meta interface{}) error
 	log.Printf("[DEBUG] List Options: %#v", listOpts)
 
 	var image images.Image
-	allPages, err := images.List(imageClient, listOpts).AllPages()
+	allPages, err := images.List(client, listOpts).AllPages()
 	if err != nil {
-		return fmt.Errorf("Unable to query images: %s", err)
+		return fmt.Errorf("unable to query images: %s", err)
 	}
 
 	allImages, err := images.ExtractImages(allPages)
 	if err != nil {
-		return fmt.Errorf("Unable to retrieve images: %s", err)
+		return fmt.Errorf("unable to retrieve images: %s", err)
+	}
+
+	var filteredImages []images.Image
+	if nameRegex, ok := d.GetOk("name_regex"); ok {
+		r := regexp.MustCompile(nameRegex.(string))
```

Comment:
It's better to validate input regex, please use `regexp.Compile`. Otherwise, this will trigger a panic for invalid regex

Reply:
name_regex has validationfunc